### PR TITLE
Webhooks fixes

### PR DESCRIPTION
**Pull request by paglias**
WIP Missing tests and new i18n strings

### CODE REVIEW

**Comment by librarianmage** [comment]
@paglias I just realized I created a similar PR ( #7622 ). Is there anything from this that would help?

**Comment by librarianmage** [comment]
The major things I noticed were:
- validator.js now only wants a string (future versions will result in an error)
- $http uses `data` instead of `body` in (website/client/js/services/userServices.js)

File: common/script/ops/addWebhook.js [hunk 1]
```diff
@@ -5,22 +5,35 @@ import {
...
 import _ from 'lodash';
+import { v4 as uuid } from 'uuid';
 
 module.exports = function addWebhook (user, req = {}) {
-  let wh = user.preferences.webhooks;
+  let webhooks = user.preferences.webhooks;
 
-  if (!validator.isURL(_.get(req, 'body.url'))) throw new BadRequest(i18n.t('invalidUrl', req.language));
-  if (!validator.isBoolean(_.get(req, 'body.enabled'))) throw new BadRequest(i18n.t('invalidEnabled', req.language));
+  let id = _.get(req, 'body.id', uuid); // Use uuid if not supplied
```
Comment (paglias) [inline common/script/ops/addWebhook.js]: _.get expects a value not a function as the third argument

File: common/script/ops/addWebhook.js [hunk 2]
```diff
@@ -5,22 +5,35 @@ import {
...
+  let enabled = _.get(req, 'body.enabled', true); // Enabled by default
+  let url = _.get(req, 'body.url');
+
+  if (!validator.isURL(id)) {
```
Comment (paglias) [inline common/script/ops/addWebhook.js]: should be removed

**Comment by crookedneighbor** [comment]
I _think_ this can be closed. If not, re-open it.

**Comment by paglias** [comment]
This has some fixes for a few bugs that still exists

**Comment by paglias** [comment]
I don't have the time to currently finish this PR :( If someone wants to take over it only tests should be missing

**Comment by crookedneighbor** [comment]
Changes from here are included in https://github.com/HabitRPG/habitrpg/pull/7879

**Pull request closed without merging**